File: NodeJsProject/src/entities/product.ts
import {
  BaseEntity,
  ManyToOne,
  JoinColumn,
  Column,
  Entity,
  PrimaryGeneratedColumn,
  OneToOne,
  ManyToMany,
  OneToMany,
} from "typeorm";
import { OrderItems } from "./purchase";
import { Store } from "./store";
import { Image } from "./photo";

@Entity("Products")
export class Product extends BaseEntity {
  @PrimaryGeneratedColumn("uuid", { name: "ProductId" })
  id: string;

  @Column({ name: "ProductName", type: "text", nullable: true })
  product_name: string;

  @Column({ type: "text", name: "ProductDescription", nullable: true })
  product_description: string;

  @Column({ type: "text", name: "ProductCategory", nullable: true })
  product_category: string;

  @Column({ type: "text", name: "UnitOfMeasurement", nullable: true })
  unit_of_measurement: string;

  @Column({ name: "Quantity" })
  quantity: number;

  @Column({ name: "Publish", nullable: false })
  publish: boolean;

  @Column({ nullable: false, name: "UnitPrice", type: "numeric" })
  unit_price: number;

  @Column({ nullable: false })
  StoreId: string;
  @ManyToOne(() => Store, (store) => store.products, {
    nullable: true,
    onDelete: "CASCADE",
  })
  @JoinColumn({
    name: "StoreId",
  })
  store: Store;

  // @OneToMany(() => OrderItems, (orderItems) => orderItems.product)
  // orderItems: OrderItems[];

  @OneToOne(() => Image, (image) => image.product)
  image: Image;
}

File: DotNetProject/Domain/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Domain;
public class Product
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Key]
    public Guid ProductId { get; set; }
	public string ProductName { get; set; }
	public string ProductDescription { get; set; }
	public string ProductCategory { get; set; }
	public string Unit { get; set; }
	public int Quantity { get; set; }
	public Guid StoreId { get; set; }
	public Store Store { get; set; }
	public List<Order> Orders { get; set; }

}
